pkg/tnf-operator/pcs/cluster.go

```diff
@@ -0,0 +1,76 @@
+package pcs
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/cluster-etcd-operator/pkg/tnf-operator/config"
+	"github.com/openshift/cluster-etcd-operator/pkg/tnf-operator/exec"
+)
+
+// ConfigureCluster checks the pcs cluster status and runs the finalization script if needed
+func ConfigureCluster(ctx context.Context, cfg config.ClusterConfig) (bool, error) {
+	klog.Info("Checking pcs cluster status")
+
+	// check how far we are...
+	stdOut, stdErr, err := exec.Execute(ctx, "/usr/sbin/pcs resource config")
+
+	if strings.Contains(stdOut, "kubelet-clone") {
+		// all done already
+		klog.Infof("HA cluster setup already done")
+		return false, nil
+	}
+
+	klog.Info("Starting initial HA cluster setup")
+
+	stonithArgsNode1 := getStonithArgs(cfg, 0)
+	stonithArgsNode2 := getStonithArgs(cfg, 1)
+
+	commands := []string{
+		fmt.Sprintf("/usr/sbin/pcs cluster setup TNF %s addr=%s %s addr=%s --debug --force", cfg.NodeName1, cfg.NodeIP1, cfg.NodeName2, cfg.NodeIP2),
```

Comment:
does this need to be executed on every node?

Reply:
> After that, things need to be executed on one node only

and you can't do this as part of the machine config itself? in a kind-of a first-one-wins manner?

I think this is also more of a job than a deployment, isn't it?

Reply:
> I think this can be a conflict at least when both nodes boot at the same time, and both try to configure pacemaker at the same time

and when you backoff and retry for a few times until it settles on a configuration? Is it really likely is it that they boot the exact same time and configure themselves _differently_ on the same nanosecond? 

> Also, fencing setup needs some credentials, which are read from a secret. I'm not sure if we can easily access the cluster from a systemd unit...

we can have the static pod controller sync that secret into the /etc/kubernetes/manifest folder if you like

Reply:
summarizing the slack thread we had:
* yes this can be done safely with a bash script in systemd
* we can ship this as part of the MCO configs
* secrets can be planted on each node via static pod controller
* how do we propagate script errors? how does kubelet do it?

Reply:
I think this looks good